src/System.Private.CoreLib/src/System/RtType.cs

```diff
@@ -3430,11 +3426,7 @@ public override string[] GetEnumNames()
             string[] ret = Enum.InternalGetNames(this);
 
             // Make a copy since we can't hand out the same array since users can modify them
-            string[] retVal = new string[ret.Length];
-
-            Array.Copy(ret, 0, retVal, 0, ret.Length);
-
-            return retVal;
+            return new ReadOnlySpan<string>(ret).ToArray();
```

Comment:
Is this just a simplification (which is good, regardless), or is there a perf benefit to this?

Reply:
Simplification. The perf benefit - if there is any - is within noise. The method is dominated by the array allocation.